js/about/preferences.js

```diff
@@ -323,6 +323,91 @@ class BitcoinDashboard extends ImmutableComponent {
   }
 }
 
+class PaymentHistory extends ImmutableComponent {
+  get ledgerData () {
+    return this.props.ledgerData
+  }
+
+  render () {
+    const transactions = this.props.ledgerData.get('transactions')
+
+    return <div id='paymentHistory'>
+      <table className='sort'>
+        <thead>
+          <tr>
+            <th className='sort-header' data-l10n-id='date' />
+            <th className='sort-header' data-l10n-id='totalAmount' />
+            <th className='sort-header' data-l10n-id='receiptLink' />
+          </tr>
+        </thead>
+        <tbody>
+        {
+          transactions.map(function (row) {
+            return <PaymentHistoryRow transaction={row} ledgerData={this.props.ledgerData} />
+          }.bind(this))
+        }
+        </tbody>
+      </table>
+    </div>
+  }
+}
+
+class PaymentHistoryRow extends ImmutableComponent {
+
+  get transaction () {
+    return this.props.transaction
+  }
+
+  get timestamp () {
+    return this.transaction.get('submissionStamp')
+  }
+
+  get formattedDate () {
+    return formattedDateFromTimestamp(this.timestamp)
+  }
+
+  get numericDateStr () {
+    var date = new Date(this.timestamp)
```

Comment:
So I did do that for the displayed date: https://github.com/brave/browser-laptop/pull/3391/commits/48b0171bb5ad37071d360358522f242ea132ce7d#diff-e3eeb751016b2ce9f8278efce585a461R1258

This is for a filename, so `toLocaleDateString` is a little awkward with the slashes and lack of zero-padding. But sure, I can use `toLocaleDateString` and then replace the '/' with '-'s and let the lack of zero-padding be.